File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/updates/UpdateStoredFiles.java
package com.lasthopesoftware.bluewater.client.stored.library.items.files.updates;

import com.lasthopesoftware.bluewater.client.browsing.items.media.files.ServiceFile;
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId;
import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile;
import com.namehillsoftware.handoff.promises.Promise;

public interface UpdateStoredFiles {
	Promise<StoredFile> promiseStoredFileUpdate(LibraryId libraryId, ServiceFile serviceFile);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/updates/UpdateStoredFiles.kt
package com.lasthopesoftware.bluewater.client.stored.library.items.files.updates

import com.lasthopesoftware.bluewater.client.browsing.items.media.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile
import com.namehillsoftware.handoff.promises.Promise

interface UpdateStoredFiles {
    fun promiseStoredFileUpdate(libraryId: LibraryId, serviceFile: ServiceFile): Promise<StoredFile?>
}
